### DISCUSSION TITLE
Flash Cart Scripts

### DISCUSSION THREAD

**Discussion by Icesythe7** [sanni/cartreader#457] [Ideas & Suggestions]
Was thinking of the idea of instead of hardcoding new repos/flash chips into the main code itself why not code them in scripts like a json file or even lua/python to be placed in a folder on the sd card, this way it is much easier to add new carts and removes the need to recompile the code with every development test and tweak and new addition in general, users could just submit a script for x flash chip and then others just toss them in the folder and nothing else is required. My idea would be the main code will look in the folder for an always present main script which will contain all the flash ids etc which will then reroute to the specific script for that chip.

**Comment by Chipchip256** [comment]
pensei na mesma ideia, porem ao fazer template individual ocasiona um delay ao executar o codigo no arduino, assim tem a leitura da ROM que esta no cartao de memoria  e outros arquivos que sao colocados na pasta raiz do cartao de memoria, nao é interessante sobrecarregar o arduino com modificações no código, mas ter o poder de personalizar a  eeprom exige muito conhecimento pra colocar cada detalhe do datasheet  para o arquivo personalizado no qual o código do arduino consegue interpretar, também tem a possibilidade de gerar um algoritmo simples ou .md que facilite a ediçao desse arquivo, sanni gostaria que voce abraçasse essa ideia e parabens por esse grande projeto, abraços do Brasil!
